usersync/cookie.go

```diff
@@ -45,26 +46,78 @@ type uidWithExpiry struct {
 }
 
 // ParsePBSCookieFromRequest parses the UserSyncMap from an HTTP Request.
-func ParsePBSCookieFromRequest(r *http.Request, cookie *config.HostCookie) *PBSCookie {
-	if cookie.OptOutCookie.Name != "" {
-		optOutCookie, err1 := r.Cookie(cookie.OptOutCookie.Name)
-		if err1 == nil && optOutCookie.Value == cookie.OptOutCookie.Value {
-			pc := NewPBSCookie()
+func ParsePBSCookieFromRequest(r *http.Request, cookieHost *config.HostCookie) *PBSCookie {
+	if cookieHost.OptOutCookie.Name != "" { //This is a cookie name, besides, "uids", stored in HostCookie
+		//Since an OptOutCookie name is defined in our server configuration, let's make sure this cookie isn't it. If it is, we return a blank cookie
+		optOutCookie, err1 := r.Cookie(cookieHost.OptOutCookie.Name) //returns a cookie inside the http.Request
+		if err1 == nil && optOutCookie.Value == cookieHost.OptOutCookie.Value {
+			var pc *PBSCookie = NewPBSCookie() //The config.HostCookie cookie was found so, ignore it? Declare a new and blank PBSCookie
 			pc.SetPreference(false)
 			return pc
 		}
 	}
+	// There's no OptOutCookie defined or at least our incoming request doesn't have that OptOutName in any http.Request.Header
+	// Let's try to download the request's cookie data into a fresh PBSCookie
 	var parsed *PBSCookie
-	uidCookie, err2 := r.Cookie(UID_COOKIE_NAME)
+	uidCookie, err2 := r.Cookie("uids") // hate the refactoring Mansi did, in real code that value is: UID_COOKIE_NAME
```

Comment:
The downside here is you now have "`uids`" defined twice. If you ever have to change it, you will need to edit two lines in this file. This can be error prone. What is the advantage of not using the `UID_COOKIE_NAME` constant?

Reply:
That was not meant to make it to the PR, I'm sorry. I don't hate the refactoring, I just found it a little confusing, let me rollback.